src/VecSim/algorithms/hnsw/hnsw.h

```diff
@@ -50,7 +50,6 @@ class HierarchicalNSW : public VecsimBaseObject {
     double epsilon_;
 
     // Index meta-data (based on the data dimensionality and index parameters)
-    size_t dim;
     size_t data_size_;
```

Comment:
We need to see which of these fields can be removed... For instance `offsetLevel0_` is always 0.

Reply:
I think we should leave this to a proper refactor PR. this value will stay in the base class of hnsw and will be simple to deal with even after the multi-value effort.